src/borg/locking.py

```diff
@@ -202,6 +202,16 @@ def break_lock(self):
                 os.unlink(os.path.join(self.path, name))
             os.rmdir(self.path)
 
+    def migrate_lock(self, old_id, new_id):
+        """migrate the lock ownership from old_id to new_id"""
+        if self.id == old_id:
```

Comment:
I don't quite see how this could be false? Perhaps make it an `assert self.id == old_id` instead?